server/src/test/java/io/crate/expression/reference/sys/snapshot/SysSnapshotsTest.java

```diff
@@ -126,6 +127,83 @@ public void testUnavailableSnapshotsAreFilteredOut() throws Exception {
         assertThat(uuids).containsExactlyInAnyOrder(s1.getUUID(), s2.getUUID(), s3.getUUID());
     }
 
+    @Test
+    public void test_global_state_and_reason_and_total_shards_are_exposed_in_sys_snapshots() throws Exception {
+        HashMap<String, SnapshotId> snapshots = new HashMap<>();
+        SnapshotId sid1 = new SnapshotId("s1", UUIDs.randomBase64UUID());
+        SnapshotId sid2 = new SnapshotId("s2", UUIDs.randomBase64UUID());
+        snapshots.put(sid1.getUUID(), sid1);
+        snapshots.put(sid2.getUUID(), sid2);
+
+        RepositoryData repositoryData = new RepositoryData(
+            1, snapshots, Collections.emptyMap(), Collections.emptyMap(), Collections.emptyMap(), ShardGenerations.EMPTY, IndexMetaDataGenerations.EMPTY);
+
+        Repository r1 = mock(Repository.class);
+        doReturn(CompletableFuture.completedFuture(repositoryData))
+            .when(r1)
+            .getRepositoryData();
+
+        when(r1.getMetadata()).thenReturn(new RepositoryMetadata("repo1", "fs", Settings.EMPTY));
+
+        Metadata metadata = mock(Metadata.class);
+        when(metadata.templates()).thenReturn(ImmutableOpenMap.of());
+
+        // s1 fails for SnapshotInfo (to verify NULL value for includeGlobalState flag).
+        doReturn(CompletableFuture.completedFuture(metadata))
+            .when(r1)
+            .getSnapshotGlobalMetadata(sid1);
+        doReturn(CompletableFuture.failedFuture(new SnapshotException("repo1", "s1", "Everything is wrong")))
+            .when(r1)
+            .getSnapshotInfo(sid1);
```

Comment:
I don't think we should use the error handling at `SysSnapshots.createSysSnapshot()` to create an instance which includes a `null` for the `includeGlobalState`. This is not really related, would be a different test case in my opinion.
Why not just return a instance of `SnapshotInfo` instead like below?

Reply:
Pushed a dedicated test for unavailable snapshot and updated existing test.

Reply:
I actually think I should keep only 1 instance of `SnapshotInfo` (with non-default values of new columns) in `test_global_state_and_reason_and_total_shards_are_exposed_in_sys_snapshots`.

There is no actual prod code returning 0 total shards and null flag for successful snapshot.